test/unittest/dds/xtypes/type_representation/TypeObjectUtilsTests.cpp

```diff
@@ -461,93 +495,398 @@ TEST(TypeObjectUtilsTests, build_plain_map_s_type_defn_inconsistencies)
+            indirect_type_ids);
+    if (EK_COMPLETE == indirect_type_ids.type_identifier1()._d())
     {
-        test_identifier = eprosima::fastcdr::external<TypeIdentifier>(new TypeIdentifier(type_ids.type_identifier2()));
+        complete_indirect_id =
+                eprosima::fastcdr::external<TypeIdentifier>(new TypeIdentifier(indirect_type_ids.type_identifier1()));
+        minimal_indirect_id =
+                eprosima::fastcdr::external<TypeIdentifier>(new TypeIdentifier(indirect_type_ids.type_identifier2()));
     }
     else
     {
-        test_identifier = eprosima::fastcdr::external<TypeIdentifier>(new TypeIdentifier(type_ids.type_identifier1()));
+        complete_indirect_id =
+                eprosima::fastcdr::external<TypeIdentifier>(new TypeIdentifier(indirect_type_ids.type_identifier2()));
+        minimal_indirect_id =
+                eprosima::fastcdr::external<TypeIdentifier>(new TypeIdentifier(indirect_type_ids.type_identifier1()));
     }
-    // TypeIdentifier inconsistent with complete header
-    EXPECT_THROW(PlainMapSTypeDefn plain_seq = TypeObjectUtils::build_plain_map_s_type_defn(
+
+    register_alias_type_object();
+    TypeIdentifierPair alias_type_ids;
+    eprosima::fastcdr::external<TypeIdentifier> complete_alias_id;
+    eprosima::fastcdr::external<TypeIdentifier> minimal_alias_id;
+    DomainParticipantFactory::get_instance()->type_object_registry().get_type_identifiers("int16_alias",
+            alias_type_ids);
+    if (EK_COMPLETE == alias_type_ids.type_identifier1()._d())
+    {
+        complete_alias_id =
+                eprosima::fastcdr::external<TypeIdentifier>(new TypeIdentifier(alias_type_ids.type_identifier1()));
+        minimal_alias_id =
+                eprosima::fastcdr::external<TypeIdentifier>(new TypeIdentifier(alias_type_ids.type_identifier2()));
+    }
+    else
+    {
+        complete_alias_id =
+                eprosima::fastcdr::external<TypeIdentifier>(new TypeIdentifier(alias_type_ids.type_identifier2()));
+        minimal_alias_id =
+                eprosima::fastcdr::external<TypeIdentifier>(new TypeIdentifier(alias_type_ids.type_identifier1()));
+    }
+
+    EXPECT_NO_THROW(PlainMapSTypeDefn plain_map = TypeObjectUtils::build_plain_map_s_type_defn(
+                fully_descriptive_header, 10, test_identifier, flags, key_identifier));
+    EXPECT_THROW(PlainMapSTypeDefn plain_map = TypeObjectUtils::build_plain_map_s_type_defn(
+                fully_descriptive_header, 10, test_identifier, flags, complete_typeid), InvalidArgumentError);
+    EXPECT_THROW(PlainMapSTypeDefn plain_map = TypeObjectUtils::build_plain_map_s_type_defn(
+                fully_descriptive_header, 10, test_identifier, flags, complete_indirect_id), InvalidArgumentError);
+    EXPECT_NO_THROW(PlainMapSTypeDefn plain_map = TypeObjectUtils::build_plain_map_s_type_defn(
+                fully_descriptive_header, 10, test_identifier, flags, complete_alias_id));
```

Comment:
I think this case should throw. Even if the alias is of a primitive type, the alias is already a TypeObject, so it is not really fully descriptive. The same applies below for the minimal alias.

Reply:
Changed aliases behabiour in [d3245e1](https://github.com/eProsima/Fast-DDS/pull/4496/commits/d3245e16dd687c62ce397be8018c718828d2bc6e)